client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx

```diff
@@ -157,79 +111,81 @@ const ScreenMain = ( {
 		elementToFocus.focus();
 	}, [ isInitialLocation ] );
 
-	useEffect( () => {
-		updateActivePatternPosition();
-	}, [ updateActivePatternPosition ] );
-
 	return (
 		<>
 			<NavigatorHeader
 				title={ <NavigatorTitle title={ translate( 'Design your own' ) } /> }
-				description={ headerDescription }
+				description={ translate(
+					'Create your homepage by first adding patterns and then choosing a color palette and font style.'
+				) }
 				hideBack
 			/>
 			<div className="screen-container__body" ref={ wrapperRef }>
 				<VStack spacing="4">
 					<NavigatorItemGroup title={ translate( 'Patterns' ) }>
 						<NavigatorItem
-							checked={ Boolean( selectedHeader ) }
+							checked={ hasHeader }
 							icon={ header }
 							aria-label={ translate( 'Header' ) }
-							onClick={ () => onMainItemSelect( { name: 'header', isPanel: true } ) }
-							active={ getIsActiveButton( 'header' ) }
+							onClick={ () => handleNavigatorItemSelect( 'header', 'header' ) }
+							active={ selectedCategory === 'header' }
 						>
 							{ translate( 'Header' ) }
 						</NavigatorItem>
 						<NavigatorItem
-							checked={ Boolean( selectedSections.length ) }
+							checked={ hasSections }
 							icon={ layout }
 							aria-label={ translate( 'Sections' ) }
-							onClick={ () => onMainItemSelect( { name: 'section', isPanel: true } ) }
-							active={ getIsActiveButton( 'section' ) }
+							onClick={ () => handleNavigatorItemSelect( 'section', CATEGORY_ALL_SLUG ) }
+							active={ shouldOpenCategoryList }
 							hasNestedItems
 						>
 							{ translate( 'Sections' ) }
 						</NavigatorItem>
 
-						{ selectedMainItem === 'section' && (
+						{ shouldOpenCategoryList && (
 							<PatternCategoryList
 								categories={ categories }
 								patternsMapByCategory={ patternsMapByCategory }
-								selectedCategory={ selectedSectionCategory }
+								selectedCategory={ selectedCategory }
 								onSelectCategory={ onSelectSectionCategory }
 							/>
 						) }
 
 						<NavigatorItem
-							checked={ Boolean( selectedFooter ) }
+							checked={ hasFooter }
 							icon={ footer }
 							aria-label={ translate( 'Footer' ) }
-							onClick={ () => onMainItemSelect( { name: 'footer', isPanel: true } ) }
-							active={ getIsActiveButton( 'footer' ) }
+							onClick={ () => handleNavigatorItemSelect( 'footer', 'footer' ) }
+							active={ selectedCategory === 'footer' }
 						>
 							{ translate( 'Footer' ) }
 						</NavigatorItem>
 					</NavigatorItemGroup>
 					<NavigatorItemGroup title={ translate( 'Styles' ) }>
 						<>
-							<NavigatorButtonAsItem
+							<NavigatorItem
```

Comment:
I want to replace the path to avoid the `goBack` behavior being weird. For example, if we pick a category and then pick Colors or Fonts, the category you selected keeps open when you finish the Colors or Fonts screen.

But this might redundant. Let me check again.